### PR TITLE
GTEST: Use more than 1 fragment in rndv ppln test

### PR DESCRIPTION
**Pull request by brminich**
## What
This test provokes a data validation failure in CI
To be extended with the fix for that failure

### CODE REVIEW

File: src/ucp/rndv/proto_rndv.c [hunk 1]
```diff
@@ -610,14 +610,11 @@ void ucp_proto_rndv_rts_abort(ucp_request_t *req, ucs_status_t status)
...
-        return UCS_OK;
+    if (req->flags & UCP_REQUEST_FLAG_PROTO_INITIALIZED) {
+        ucs_assert(req->send.state.completed_size == 0);
     }
 
-    ucs_assert(req->send.state.completed_size == 0);
-    ucp_send_request_id_release(req);
-    ucp_proto_request_zcopy_clean(req, UCP_DT_MASK_ALL);
-    return UCS_OK;
+    return ucp_proto_request_zcopy_id_reset(req);
```
Comment (yosefe) [inline src/ucp/rndv/proto_rndv.c:617]: nice catch and fix!

File: test/gtest/ucp/test_ucp_am.cc [hunk 2]
```diff
@@ -1848,7 +1849,9 @@ class test_ucp_am_nbx_rndv_memtype : public test_ucp_am_nbx_rndv {
 
 UCS_TEST_P(test_ucp_am_nbx_rndv_memtype, rndv)
 {
-    test_am_send_recv_memtype(64 * UCS_KBYTE);
+    ucp_context_config_t *cfg = &sender().worker()->context->config.ext;
+    size_t size               = cfg->rndv_frag_size[UCS_MEMORY_TYPE_HOST] + 1;
+    test_am_send_recv_memtype(size);
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:1854]: maybe worth testing several sizes that are "interesting"?

**Review by yosefe — COMMENTED** [review]
@evgeny-leksikov can you pls review as well?

File: test/gtest/ucp/test_ucp_am.cc (hunk 2 above)
Comment (brminich) [reply test/gtest/ucp/test_ucp_am.cc:1854]: can you sugget some?
this test has many variants, i'd try not to add a lot to keep CXI time reasonable

File: src/ucp/rndv/rndv_rtr.c [hunk 3]
```diff
@@ -301,11 +301,9 @@ ucp_proto_rndv_rtr_mtype_abort(ucp_request_t *req, ucs_status_t status)
...
 static ucs_status_t ucp_proto_rndv_rtr_mtype_reset(ucp_request_t *req)
 {
-    ucs_mpool_put_inline(req->send.rndv.mdesc);
-    if (ucp_proto_rndv_request_is_ppln_frag(req)) {
-        req->status = UCS_ERR_CANCELED;
-        ucp_proto_rndv_ppln_recv_frag_clean(req);
-        return UCS_ERR_CANCELED;
+    if (req->flags & UCP_REQUEST_FLAG_PROTO_INITIALIZED) {
+        ucs_mpool_put_inline(req->send.rndv.mdesc);
+        req->send.rndv.mdesc = NULL;
```
Comment (evgeny-leksikov) [inline src/ucp/rndv/rndv_rtr.c:306]: should not we propagate status to super recv request and reset it too?

File: test/gtest/ucp/test_ucp_am.cc (hunk 2 above)
Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_am.cc:1854]: > can you sugget some?

at least do not remove old values, usually init/cleanup takes significat part of time in comparison with send/recv

Comment (brminich) [reply test/gtest/ucp/test_ucp_am.cc:1854]: do you mean to also test 64kb as before?

File: src/ucp/rndv/rndv_rtr.c (hunk 3 above)
Comment (brminich) [reply src/ucp/rndv/rndv_rtr.c:306]: this is a reset function, why is status relevant here? Do you mean a user receive request? Not sure, it's not supposed to change afaiu

File: test/gtest/ucp/test_ucp_am.cc (hunk 2 above)
Comment (yosefe) [reply test/gtest/ucp/test_ucp_am.cc:1854]: yes: 64k, frag_size+1, frage_size-1

File: src/ucp/rndv/rndv_rtr.c (hunk 3 above)
Comment (evgeny-leksikov) [reply src/ucp/rndv/rndv_rtr.c:306]: ok

File: test/gtest/ucp/test_ucp_am.cc (hunk 2 above)
Comment (brminich) [reply test/gtest/ucp/test_ucp_am.cc]: updated + squashed, because the PR is small

File: test/gtest/ucp/test_ucp_am.cc [hunk 4]
```diff
@@ -1848,7 +1849,13 @@ class test_ucp_am_nbx_rndv_memtype : public test_ucp_am_nbx_rndv {
 
 UCS_TEST_P(test_ucp_am_nbx_rndv_memtype, rndv)
 {
-    test_am_send_recv_memtype(64 * UCS_KBYTE);
+    ucp_context_config_t *cfg = &sender().worker()->context->config.ext;
+    size_t size               = cfg->rndv_frag_size[UCS_MEMORY_TYPE_HOST];
+    size_t sizes[]            = {64 * UCS_KBYTE, size - 1, size + 1};
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_am.cc:1854]:
```
const auto cfg = ...
const size_t rndv_frag_size = ...
const std::vector<size_t> sizes = {...}

for (size_t size : sizes) {
  test_am ...
}
```

**Review by evgeny-leksikov — DISMISSED** [review]
taking in account @yosefe 's c++ code style suggestion, LGTM

File: test/gtest/ucp/test_ucp_am.cc [hunk 5]
```diff
@@ -1848,7 +1849,14 @@ class test_ucp_am_nbx_rndv_memtype : public test_ucp_am_nbx_rndv {
 
 UCS_TEST_P(test_ucp_am_nbx_rndv_memtype, rndv)
 {
-    test_am_send_recv_memtype(64 * UCS_KBYTE);
+    const auto *cfg                 = &sender().worker()->context->config.ext;
+    const size_t rndv_frag_size     = cfg->rndv_frag_size[UCS_MEMORY_TYPE_HOST];
```
Comment (evgeny-leksikov) [inline test/gtest/ucp/test_ucp_am.cc:1853]: minor:
```suggestion
    const auto &cfg                 = sender().worker()->context->config.ext;
    const size_t rndv_frag_size     = cfg.rndv_frag_size[UCS_MEMORY_TYPE_HOST];
```

Comment (brminich) [reply test/gtest/ucp/test_ucp_am.cc:1853]: i'd keep it as is to avoid restarting CI if you don't mind

Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_am.cc:1853]: > i'd keep it as is to avoid restarting CI if you don't mind

ok

**Review by yosefe — APPROVED** [review]

**Pull request merged**